src/main/java/de/tum/in/www1/artemis/service/FilePathService.java

```diff
@@ -20,43 +27,211 @@ public void setFileUploadPathStatic(String fileUploadPath) {
+        return Path.of(fileUploadPath, "images", "course", "icons");
     }
 
-    public static String getDragAndDropBackgroundFilePath() {
-        return Path.of(fileUploadPath, "images", "drag-and-drop", "backgrounds").toString();
+    public static Path getExamUserSignatureFilePath() {
+        return Path.of(fileUploadPath, "images", "exam-user", "signatures");
     }
 
-    public static String getDragItemFilePath() {
-        return Path.of(fileUploadPath, "images", "drag-and-drop", "drag-items").toString();
+    public static Path getStudentImageFilePath() {
+        return Path.of(fileUploadPath, "images", "exam-user");
     }
 
-    public static String getCourseIconFilePath() {
-        return Path.of(fileUploadPath, "images", "course", "icons").toString();
+    public static Path getLectureAttachmentFilePath() {
+        return Path.of(fileUploadPath, "attachments", "lecture");
     }
 
-    public static String getExamUserSignatureFilePath() {
-        return Path.of(fileUploadPath, "images", "exam-user", "signatures").toString();
+    public static Path getAttachmentUnitFilePath() {
+        return Path.of(fileUploadPath, "attachments", "attachment-unit");
     }
 
-    public static String getStudentImageFilePath() {
-        return Path.of(fileUploadPath, "images", "exam-user").toString();
+    public static Path getFileUploadExercisesFilePath() {
+        return Path.of(fileUploadPath, "file-upload-exercises");
     }
 
-    public static String getLectureAttachmentFilePath() {
-        return Path.of(fileUploadPath, "attachments", "lecture").toString();
+    public static Path getMarkdownFilePath() {
+        return Path.of(fileUploadPath, "markdown");
     }
 
-    public static String getAttachmentUnitFilePath() {
-        return Path.of(fileUploadPath, "attachments", "attachment-unit").toString();
+    /**
+     * Convert the given public file url to its corresponding local path
+     *
+     * @param publicPath the public file url to convert
+     * @throws FilePathParsingException if the path is unknown
+     * @return the actual path to that file in the local filesystem
+     */
+    public Path actualPathForPublicPathOrThrow(URI publicPath) {
+        Path actualPath = actualPathForPublicPath(publicPath);
+        if (actualPath == null) {
+            // path is unknown => cannot convert
+            throw new FilePathParsingException("Unknown Filepath: " + publicPath);
+        }
+
+        return actualPath;
+    }
+
+    /**
+     * Convert the given public file url to its corresponding local path
+     *
+     * @param publicPath the public file url to convert
+     * @return the actual path to that file in the local filesystem
+     */
+    public Path actualPathForPublicPath(URI publicPath) {
+        // first extract the filename from the url
+        String uriPath = publicPath.getPath();
+        Path path = Path.of(uriPath);
+        String filename = uriPath.substring(uriPath.lastIndexOf('/') + 1);
```

Comment:
```suggestion
        String filename = path.getFileName().toString();
```
Would this be identical?

Reply:
Yes!